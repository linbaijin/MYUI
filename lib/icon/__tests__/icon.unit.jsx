import render from 'react-test-renderer'
import React from 'react'
import Icon from '../icon'
import {mount} from 'enzyme'


describe('Icon',()=>{
    it('xxx',()=>{
        const json = render.create(<Icon name="alipay"/>).toJSON()
        expect(json).toMatchSnapshot()
    })
    it('onClick',()=>{
        const fn = jest.fn()
        const fn2 = jest.fn()
        const component = mount(<Icon name="alipay" onClick={fn2}/>)
        component.find('svg').simulate('click')
        expect(fn2).toBeCalled()
    })
})