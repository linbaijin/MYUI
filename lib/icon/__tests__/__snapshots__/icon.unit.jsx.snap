// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon xxx 1`] = `
<svg
  className="icon"
  fill="currentcolor"
>
  <use
    xlinkHref="#alipay"
  />
</svg>
`;
